configure.ac

```diff
@@ -2322,6 +2322,15 @@ AH_BOTTOM([
 # endif
 #endif
 
+/* time.h declares a `timezone` global which causes stupid problems
+ * when we accidentally use it as a local name too, but mess up the
+ * scoping.  So, make sure we can not use it as a local name too,
+ * except very carefully!
+ */
+#ifndef REALLY_USE_LIBC_TIMEZONE
+#define timezone /* dont call things this! */
```

Comment:
You can even do this:

```
#define timezone "Don't use timezone as a variable name"
```



Which leads to:
```
In file included from imap/jmap_ical.c:44:
imap/jmap_ical.c: In function ‘location_from_ical’:
./config.h:781:18: error: expected identifier or ‘(’ before string constant
  781 | #define timezone "Don't use timezone as a variable name"
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
./config.h:781:18: note: in definition of macro ‘timezone’
  781 | #define timezone "Don't use timezone as a variable name"
      |                  ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

Reply:
oh that's amazing, thank you

Reply:
Hmm.  I tried that out by undoing the change in #3945, and it wouldn't have caught that case! :(

The problem there was:

```
     if (timezone /* whoops, wrong name! */) {
```

and it slides by unremarked because a string constant is true in a boolean context.